src/thread_manager.ts

```diff
@@ -10,20 +9,25 @@ const MAX_QUERY_THREADS_LIMIT = 25;
 
 export type ThreadManagerState<SCG extends ExtendableGenerics = DefaultGenerics> = {
   active: boolean;
-  existingReorderedThreadIds: string[];
-  lastConnectionDownAt: Date | null;
-  loadingNextPage: boolean;
+  isThreadOrderStale: boolean;
+  lastConnectionDropAt: Date | null;
+  pagination: ThreadManagerPagination;
+  ready: boolean;
```

Comment:
Is this a direct replacement of a
```ts
if (!threads.length && nextCursor !== null) return;
```
in new message handler? I'm not sure if extra property is needed only for this case, the value could be derived from the existing values. Addin this also means that there's an extra one to take care of manually.